variables.tf

```diff
@@ -202,3 +202,46 @@ variable "suffix" {
     error_message = "If the suffix value is set then it must be at least 4 characters long."
   }
 }
+
+variable "use_existing_iam_role" {
+  type        = bool
+  default     = false
+  description = "Set this to true to use an existing IAM role"
+}
+
+variable "use_existing_task_role" {
+  type        = bool
+  default     = false
+  description = "Set this to true to use an existing IAM task role"
+}
+
+variable "use_existing_execution_role" {
+  type        = bool
+  default     = false
+  description = "Set this to true to use an existing IAM execution role"
+}
+
+variable "use_existing_event_role" {
+  type        = bool
+  default     = false
+  description = "Set this to true to use an existing IAM event role"
+}
+
+variable "iam_role_arn" {
+  type        = string
+  default     = ""
+  description = "The IAM role ARN is required when setting use_existing_iam_role to true"
+}
+
+variable "iam_role_name" {
+  type        = string
+  default     = ""
+  description = "The IAM role name. Required to match with iam_role_arn if use_existing_iam_role is set to true"
+}
```

Comment:
Thoughts on calling these something more specific like `cross_account_iam_role_arn` and `cross_account_iam_role_name`?

Pro: specific is better
Con: it might not match our other Terraform so lack of consistency

Considering those, I'd vote for specific over consistency just because in this case we have several roles at play.

Reply:
Yeah I thought about this. Our other TF code for CloudTrail uses iam_role_arn and iam_role_name, so that was why I used it. But I'd vote for specificity as well.